.github/workflows/remote-build-android-hybrid.yml

```diff
@@ -0,0 +1,30 @@
+name: Remote Build Android Hybrid
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
```

Comment:
wondering if we can/should be running this on push to any branch?

I don't think that would cause builds to happen when code is pushed to branches in a fork (which I'm not sure we want to pay for),  but if we remove the `branches` filter and instead do `branches-ignore: [staging, production]`, then if any Expensify engineer pushes a branch that changes native code, a C+ or other Expensify engineer could checkout that branch and get a fast build.

Reply:
You're correct and we definitely can do that. This is in fact the ultimate setup we can aim for to maximize cache hits. We can always revert back to any other strategy if we see it's too much.

- It won't run for forks as they live separately
- This will speed up feature development and testing, exactly as you mentioned. If one feature dev pushes a native update, all the other developers working on this feature will be able to use the cached one 🙂
- For forks I think we'll still have manual triggers internal devs could run it they want to

I'll push this update now.

Reply:
Ideally we want this workflow to be fast and be ran often, with vast majority (90%+) being cache hits. 
As a followup I'll be also looking at how we can speed up setting up Node so we can squeeze every second of this workflow.

Reply:
Think about it as _React's re-rendering model on the CI_: we build everything from scratch so you don’t have to overthink the caches. When RNEF detects there’s a cache hit, it will proceed with its' optimizations.